astrobee/survey_manager/scripts/command_astrobee

```diff
@@ -0,0 +1,152 @@
+
+    try:
+        while process.poll() is None:
+
+            # Get user input dynamically
+            user_input = input()
+
+            # Check if the user wants to exit
+            if user_input.lower().strip() == 'exit':
+                break
+
+            # Send the user input to the process
+            process.stdin.write(user_input + "\n")
+            process.stdin.flush()
+
+        # Close stdin to indicate no more input will be sent
+        process.stdin.close()
+
+        # Read output and error (if any)
+        output, error = process.communicate()
+
+        # Get the final exit code
+        return  process.returncode
+
+    except:
+        return -1
+
+
+def send_command(command):
+    print(command)
+    process = subprocess.Popen(command, shell=True, stdout=sys.stdout, stderr=sys.stdout, text=True)
+
+    # Get the output and error (if any)
+    return process.wait()
+
+def get_position(bay):
+
+    if bay == "jem_bay1":
+        return "'11, -4, 4.8'"
+    if bay == "jem_bay2":
+        return "'11, -5, 4.8'"
+    if bay == "jem_bay3":
+        return "'11, -6, 4.8'"
+    if bay == "jem_bay4":
+        return "'11, -7, 4.8'"
+    if bay == "jem_bay5":
+        return "'11, -8, 4.8'"
+    if bay == "jem_bay6":
+        return "'11, -9, 4.8'"
+    if bay == "jem_bay7":
+        return "'11, -9.7, 4.8'"
+
+
+def repeat_inspection():
+    while True:
+        user_input = input("Do you want to continue? (y/n): ").lower().strip()
+        if user_input == 'y':
+            if send_command_with_input("rosrun inspection inspection_tool -resume") != 0:
+                repeat_inspection()  # Continue recursively
+            break
+        elif user_input == 'n':
+            print("Exiting.")
+            break
+        else:
+            print("Invalid input. Please enter 'y' or 'n'.")
+
+
+def survey_manager_executer(goal, bay, run):
+
+    if goal == "dock":
+        send_command("rosrun executive teleop -dock")
+
+    elif goal == "dock":
+        send_command("rosrun executive teleop -undock")
+
+    elif goal == "move":
+        send_command("rosrun executive teleop -move -pos " + bay)
+
+    elif goal == "panorama":
+        send_command("python gds_helper_batch.py -i cmd -- bagger -start pano_" + bay + "_" + run)
+        if send_command_with_input("rosrun inspection inspection_tool -panorama -panorama_mode '5_mapper_and_hugin' -pos " + str(get_position(bay))) != 0:
+            repeat_inspection()
+        send_command("python gds_helper_batch.py -i cmd -- bagger -stop")
+
+    elif goal == "stereo":
+        send_command("python gds_helper_batch.py -i cmd -- bagger -start stereo_" + bay + "_" + run)
+        send_command("python gds_helper_batch.py -i cmd -- plan -load plans/ISAAC/" + bay + "_stereo_mapping.fplan")
+        send_command("python gds_helper_batch.py -i cmd -- plan -run")
+        send_command("python gds_helper_batch.py -i cmd -- bagger -stop")
```

Comment:
I know this is a work in progress, just wanted to re-raise some issues we've discussed that need resolution;

- Propagating errors up from these subcommands to represent action failure (may be a good place to use Python exceptions)
- But probably good to have a default interactive retry loop around most commands (continues on success, waits for operator feedback about retry on failure, and if you choose retry the plansys2 executor is never told about the failure). For the multi-subcommand actions, it's an interesting question whether the retry should wrap around the whole sequence or the individual subcommands.
- Some subcommands (`bagger -stop`) sort of want to be in a Python `finally:` clause so they always get run even if an earlier command fails.
- Multi-robot will affect this somehow. Probably either wrapping other-robot commands in `ssh` or somehow namespacing the ROS topics.